tables/Tables/Table.h

```diff
@@ -1035,45 +1035,50 @@ friend class TableExprNodeRep;
     // </group>
 
 protected:
-    BaseTable*  baseTabPtr_p;                 //# ptr to table representation
-    //# The isCounted_p flag is normally true.
-    //# Only for internally used Table objects (i.e. in the DataManager)
-    //# this flag is False, otherwise a mutual dependency would exist.
-    //# The DataManager has a Table object, which gets deleted by the
-    //# DataManager destructor. The DataManager gets deleted by the
-    //# PlainTable destructor, which gets called when the last Table
-    //# object gets destructed. That would never be the case if this
-    //# internally used Table object was counted.
-    Bool        isCounted_p;
-    //# Counter of last call to hasDataChanged.
+    // Shared pointer to count the references to the BaseTable object.
+    // The shared pointer can be null, so it is not counted which is necessary for
+    // the Table object in the DataManager. Otherwise mutual referencing would occur.
+    // Note that the BaseTable object contains a weak_ptr to itself which is the
+    // basis for all shared pointer counting.
+    std::shared_ptr<BaseTable> countedTabPtr_p;
+    // Pointer to BaseTable object which is always filled and always used.
+    // The shared_ptr above is only for reference counting.
+    BaseTable* baseTabPtr_p;
```

Comment:
Now using std::enable_shared_from_this instead opf own wek_ptr in BaseTable.h
However, the separate baseTabPtr_p is needed, because not all BaseTable instances can be counted due to mutual referencing as a Table object is used in DataManager.
So baseTabPtr_p is used in all functions, while countedTabPtr_p is only used for counting the references and for the final automatic deletion of BaseTable.